src/channel.cr

```diff
@@ -41,10 +57,18 @@ abstract class Channel(T)
     @closed
   end
 
+  # Receive a value from the channel.
+  #
+  # ```
+  # channel = Channel(Int32).new
+  # channel.send(1)
+  # channel.receive # => 1
+  # ```
   def receive
     receive_impl { raise ClosedError.new }
   end
 
+  # Receive a value from the channel, if any.
```

Comment:
The difference between `receive` and `receive?` is that `receive?` will not raise an exception if the channel is closed. Other `Channel` methods will raise exceptions if it is closed; I recommend reading [`channel_spec.cr`](https://github.com/crystal-lang/crystal/blob/master/spec/std/channel_spec.cr) and grep for `expect_raises` to better understand when this is expected to occur, and maybe mention this in some docs ("Raises `Channel::ClosedError` if the channel is closed", etc..)

Reply:
Yes, raised exceptions and other error behaviour (like returning `nil`) should be explicitly mentioned.

```suggestion
  # Receives a value from the channel.
  # If there is a value waiting, it is returned immediately. Otherwise, this method blocks until a value is sent to the channel.
  # 
  # Returns `nil` if the channel is closed or closes while waiting for receive.
```

Ditto for `#send`.